recipes/opencv/4.x/conanfile.py

```diff
@@ -289,6 +291,9 @@ def _configure_cmake(self):
         self._cmake.definitions["WITH_MSMF"] = self.settings.compiler == "Visual Studio"
         self._cmake.definitions["WITH_MSMF_DXVA"] = self.settings.compiler == "Visual Studio"
         self._cmake.definitions["OPENCV_MODULES_PUBLIC"] = "opencv"
+        
+        if self.options.detect_cpu_baseline:
+            self._cmake.definitions["CPU_BASELINE"] = "DETECT"
```

Comment:
I'm wondering, shouln't this always be set to `DETECT`?
Using `NATIVE`/`HOST` is not ok, since it will create a build system dependent library.

Reply:
Either DETECT or empty, but I am not fully aware of the consequences. Maybe  -DCPU_BASELINE=AVX2;MMX is also valid?

Reply:
Indeed, it's a multi-valued variable. Its usage looks complex.

I'm trying to avoid a situation in which we add an option that might complicate adding options later.
Is `CPU_BASELINE=DETECT;AVX2;MMX` a valid value?

Reply:
no Idea how it evaluates this: `elseif(";${CPU_BASELINE};" MATCHES ";DETECT;")`
cmake magic 🧙

Reply:
It's a regex. It's true when the string contains `;DETECT;`.
This also means that the string `DETECT` will not match the condition.
It's an awkward interface. My guess it's a remainder of autotools.

Reply:
It will detect the string `DETECT`, see the `;` around `${CPU_BASELINE}`?

Reply:
Yes yes, I see. My mistake!